[
    {
        "content": "<p>Hello,<br>\nI fail to have my LDAP sync.<br>\nI have a Zulip server (docker) with EmailAuthBackend and ZulipLDAPAuthBackend<br>\nI have users created by that both channels</p>\n<p>I have users in LDAP. These users still remain in LDAP whatever happens.<br>\nI filter users I want to have in zulip with that filter<br>\nLDAPSearch(\\\"ou=users,#LDAP_BASE#\\\", ldap.SCOPE_SUBTREE, \\\"(&amp;(&amp;(|(objectclass=inetOrgPerson))(|(memberof=cn=mygroup,ou=groups,#LDAP_BASE#)))(uid=%(user)s))\\\")<br>\nIt works, users can connect.</p>\n<p>Issue is coming when users are removed from mygroup.</p>\n<p>With params<br>\n     SETTING_LDAP_APPEND_DOMAIN: \"None\"<br>\n     SETTING_LDAP_DEACTIVATE_NON_MATCHING_USERS: \"False\"<br>\n     SETTING_NAME_CHANGES_DISABLED: \"True\"<br>\nwhen I do 'manage.py sync_ldap_user_data' it gives<br>\n2022-11-01 18:33:15.534 ERR  [zulip.sync_ldap_user_data] Error attempting to update user <a href=\"mailto:toto@mydom.com\">toto@mydom.com</a>:<br>\n2022-11-01 18:33:15.536 ERR  [zulip.sync_ldap_user_data] populate_user unexpectedly returned None<br>\nI try deactivate concerned users, but without any success.</p>\n<p>With params<br>\n     SETTING_LDAP_APPEND_DOMAIN: \"None\"<br>\n     SETTING_LDAP_DEACTIVATE_NON_MATCHING_USERS: \"True\"<br>\n     SETTING_NAME_CHANGES_DISABLED: \"True\"<br>\nwhen I do 'manage.py sync_ldap_user_data' it gives<br>\n2022-11-01 18:57:45.314 ERR  [zulip.sync_ldap_user_data] LDAP sync failed<br>\nTraceback (most recent call last):<br>\n  File \"/home/zulip/deployments/2022-08-24-23-56-24/zproject/backends.py\", line 1096, in sync_user_from_ldap<br>\n    ldap_username = backend.django_to_ldap_username(user_profile.delivery_email)<br>\n  File \"/home/zulip/deployments/2022-08-24-23-56-24/zproject/backends.py\", line 665, in django_to_ldap_username<br>\n    raise ZulipLDAPExceptionNoMatchingLDAPUser(<br>\nzproject.backends.ZulipLDAPExceptionNoMatchingLDAPUser: No LDAP user matching django_to_ldap_username result: <a href=\"mailto:notldapuser@mydom.tld\">notldapuser@mydom.tld</a></p>\n<p>Thus, I don't know to solve that situation !?</p>",
        "id": 1458011,
        "sender_full_name": "FrancoisPE",
        "timestamp": 1667329179
    },
    {
        "content": "<p>This topic was moved here from <a class=\"stream-topic\" data-stream-id=\"9\" href=\"/#narrow/stream/9-issues/topic/users.20removed.20from.20LDAP.20and.20sync\">#issues &gt; users removed from LDAP and sync</a> by <span class=\"user-mention silent\" data-user-id=\"19257\">Alya Abbott</span>.</p>",
        "id": 1458036,
        "sender_full_name": "Notification Bot",
        "timestamp": 1667339806
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"23927\">@FrancoisPE</span> I think we've fixed this bug relatively recently but haven't backported the fix; I would recommend upgrading to a 6.0 beta. See <a class=\"stream-topic\" data-stream-id=\"1\" href=\"/#narrow/stream/1-announce/topic/server.20releases\">#announce &gt; server releases</a> for details. (There should be a next beta posted any day now).</p>",
        "id": 1458218,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1667368546
    },
    {
        "content": "<p>Thanks a lot for your answer<br>\nDo you have an odea of keywords to find that issue resolution in changelog ?</p>",
        "id": 1458249,
        "sender_full_name": "FrancoisPE",
        "timestamp": 1667385056
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"23927\">@FrancoisPE</span>the thing we fixed was <a href=\"https://github.com/zulip/zulip/commit/a94b2572beff2cd5802179f630c5d487879be870\">a94b2572beff2cd5802179f630c5d487879be870</a> / <a href=\"https://github.com/zulip/zulip/pull/21709\">#21709</a>; so you can try applying that patch to your instance and see if it helps.</p>",
        "id": 1458285,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1667406533
    },
    {
        "content": "<p>Thanks a lot <span aria-label=\"big smile\" class=\"emoji emoji-1f604\" role=\"img\" title=\"big smile\">:big_smile:</span></p>",
        "id": 1458466,
        "sender_full_name": "FrancoisPE",
        "timestamp": 1667420999
    }
]